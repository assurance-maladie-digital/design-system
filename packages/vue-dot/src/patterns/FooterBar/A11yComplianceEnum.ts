export enum A11yComplianceEnum {
	NON_COMPLIANT = 'non-compliant',
	PARTIALLY_COMPLIANT = 'partially-compliant',
	FULLY_COMPLIANT = 'fully-compliant'
}

export const A11Y_COMPLIANCE_ENUM_VALUES = Object.values(A11yComplianceEnum);
