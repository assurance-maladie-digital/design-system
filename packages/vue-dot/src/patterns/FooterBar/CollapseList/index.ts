import CollapseList from './CollapseList.vue';

export default CollapseList;
