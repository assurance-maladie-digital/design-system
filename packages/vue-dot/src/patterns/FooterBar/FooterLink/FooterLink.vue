<template>
	<div class="d-flex">
		<a
			v-if="sitemapExternalLink || cguExternalLink || a11yStatementExternalLink || legalNoticeExternalLink"
			v-bind="options"
			:href="sitemapExternalLink || cguExternalLink || a11yStatementExternalLink || legalNoticeExternalLink"
			target="_blank"
		>
			<slot />
		</a>
		<RouterLink
			v-else
			v-bind="options"
			:to="sitemapRoute || cguRoute || a11yStatementRoute || legalNoticeRoute"
		>
			<slot />
		</RouterLink>
	</div>
</template>

<script lang="ts">
	import Vue, { PropType } from 'vue';
	import Component, { mixins } from 'vue-class-component';

	import { RawLocation } from 'vue-router';

	const Props = Vue.extend({
		props: {
			options: {
				type: Object as PropType<Record<string, unknown>>,
				default: () => ({})
			},
			sitemapRoute: {
				type: [Array, Object, String] as PropType<RawLocation>,
				default: () => ({ name: 'sitemap' })
			},
			sitemapExternalLink: {
				type: String as PropType<string>,
				default: null
			},
			cguRoute: {
				type: [Array, Object, String] as PropType<RawLocation>,
				default: () => ({ name: 'cgu' })
			},
			cguExternalLink: {
				type: String as PropType<string>,
				default: null
			},
			legalNoticeRoute: {
				type: [Array, Object, String] as PropType<RawLocation>,
				default: () => ({ name: 'legalNotice' })
			},
			legalNoticeExternalLink: {
				type: String as PropType<string>,
				default: null
			},
			a11yStatementRoute: {
				type: [Array, Object, String] as PropType<RawLocation>,
				default: () => ({ name: 'a11yStatement' })
			},
			a11yStatementExternalLink: {
				type: String as PropType<string>,
				default: null
			}
		}
	});

	const MixinsDeclaration = mixins(Props);

	@Component
	export default class FooterLink extends MixinsDeclaration {}
</script>
