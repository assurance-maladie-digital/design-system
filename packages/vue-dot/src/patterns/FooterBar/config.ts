export const config = {
	footer: {
		elevation: 3,
		color: '#fff',
		minHeight: '40px'
	},
	goTopBtn: {
		small: true,
		icon: true,
		color: 'primary'
	}
};
