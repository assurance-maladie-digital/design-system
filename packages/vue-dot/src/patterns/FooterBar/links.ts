import { mdiTwitter, mdiLinkedin } from '@mdi/js';

import { SocialMediaLink } from './SocialMediaLinks/types';

export const links: SocialMediaLink[] = [
	{
		icon: mdiLinkedin,
		href: 'https://www.linkedin.com/company/assurance-maladie/'
	},
	{
		icon: mdiTwitter,
		href: 'https://twitter.com/Assur_Maladie'
	}
];
