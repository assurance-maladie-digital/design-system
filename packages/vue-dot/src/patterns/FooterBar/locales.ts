import { A11yComplianceEnum } from './A11yComplianceEnum';

export const locales = {
	sitemapLabel: 'Plan du site',
	cguLabel: 'Conditions générales d’utilisation',
	legalNoticeLabel: 'Mentions légales',
	versionLabel: 'Version',
	[A11yComplianceEnum.NON_COMPLIANT]: 'non conforme',
	[A11yComplianceEnum.PARTIALLY_COMPLIANT]: 'partiellement conforme',
	[A11yComplianceEnum.FULLY_COMPLIANT]: 'totalement conforme',
	a11yLabel: (complianceLabel: string): string => `Accessibilité\xa0: ${complianceLabel}`
};
