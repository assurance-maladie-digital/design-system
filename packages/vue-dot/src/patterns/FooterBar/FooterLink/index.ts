import FooterLink from './FooterLink.vue';

export default FooterLink;
