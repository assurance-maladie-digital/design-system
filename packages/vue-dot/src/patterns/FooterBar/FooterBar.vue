<template>
	<VFooter
		v-bind="{
			...options.footer,
			...$attrs
		}"
		:class="{ 'py-4 py-sm-7 px-4 px-md-14': extendedMode }"
		role="contentinfo"
		class="vd-footer-bar flex-column align-stretch pa-3 w-100"
	>
		<div
			v-if="extendedMode"
			class="d-flex align-start align-sm-center mb-6"
		>
			<div class="d-flex flex-grow-1 flex-column flex-sm-row">
				<slot name="logo">
					<Logo
						v-if="!hideLogo"
						:size="logoSize"
						:class="{ 'mb-2 mb-sm-0': !hideSocialMediaLinks }"
					/>
				</slot>

				<VSpacer v-bind="options.spacer" />

				<slot name="social-media-links">
					<SocialMediaLinks
						v-if="!hideSocialMediaLinks"
						:links="socialMediaLinks"
						class="mr-8"
					/>
				</slot>
			</div>

			<VBtn
				v-bind="options.goTopBtn"
				:aria-label="locales.goTopBtnLabel"
				@click="scrollToTop"
			>
				<VIcon v-bind="options.goTopBtnIcon">
					{{ arrowTopIcon }}
				</VIcon>
			</VBtn>
		</div>

		<VDivider
			v-if="extendedMode"
			v-bind="options.divider"
			class="mb-3"
		/>

		<slot />

		<VDivider
			v-if="extendedMode"
			v-bind="options.divider"
			class="mt-3 mb-6"
		/>

		<div
			:class="{ 'py-2 py-sm-0': !extendedMode }"
			class="vd-footer-bar-links text-sm-center d-flex flex-column flex-sm-row flex-wrap align-start justify-center max-width-none mx-n3 my-n3"
		>
			<slot name="prepend" />

			<component
				:is="getLinkComponent(item)"
				v-for="(item, index) in footerLinksMapping"
				:key="index"
				:href="item.href"
				:to="item.to"
				:aria-label="item.ariaLabel"
				:target="item.openInNewTab ? '_blank' : undefined"
				:rel="item.openInNewTab ? 'noopener noreferrer' : undefined"
				class="text--primary my-3 mx-4"
			>
				{{ item.text }}
			</component>

			<p
				v-if="version"
				class="text--secondary my-3 mx-4"
			>
				{{ locales.versionLabel }} {{ version }}
			</p>

			<slot name="append" />
		</div>
	</VFooter>
</template>

<script lang="ts">
	import Vue, { PropType } from 'vue';
	import Component, { mixins } from 'vue-class-component';

	import { RawLocation } from 'vue-router';

	import { LogoSizeEnum } from '../../elements/Logo/LogoSizeEnum';

	import SocialMediaLinks from './SocialMediaLinks';
	import { SocialMediaLink } from './SocialMediaLinks/types';

	import { config } from './config';
	import { locales } from './locales';
	import { LinkItem } from './types';
	import { defaultSocialMediaLinks } from './defaultSocialMediaLinks';
	import { A11yComplianceEnum, A11Y_COMPLIANCE_ENUM_VALUES } from './A11yComplianceEnum';

	import { propValidator } from '../../helpers/propValidator';

	import { customizable } from '../../mixins/customizable';

	import { mdiArrowUp } from '@mdi/js';

	const Props = Vue.extend({
		props: {
			a11yCompliance: {
				type: String as PropType<A11yComplianceEnum>,
				default: A11yComplianceEnum.NON_COMPLIANT,
				validator: (value: A11yComplianceEnum) => propValidator('a11y-compliance', A11Y_COMPLIANCE_ENUM_VALUES, value)
			},
			linkItems: {
				type: [Array] as PropType<LinkItem[]>,
				default: null
			},
			sitemapRoute: {
				type: [Array, Object, String] as PropType<RawLocation>,
				default: () => ({ name: 'sitemap' })
			},
			cguRoute: {
				type: [Array, Object, String] as PropType<RawLocation>,
				default: () => ({ name: 'cgu' })
			},
			cookiesRoute: {
				type: [Array, Object, String] as PropType<RawLocation>,
				default: () => ({ name: 'cookies' })
			},
			legalNoticeRoute: {
				type: [Array, Object, String] as PropType<RawLocation>,
				default: () => ({ name: 'legalNotice' })
			},
			a11yStatementRoute: {
				type: [Array, Object, String] as PropType<RawLocation>,
				default: () => ({ name: 'a11yStatement' })
			},
			hideSitemapLink: {
				type: Boolean,
				default: false
			},
			hideCguLink: {
				type: Boolean,
				default: false
			},
			hideCookiesLink: {
				type: Boolean,
				default: false
			},
			hideLegalNoticeLink: {
				type: Boolean,
				default: false
			},
			hideA11yLink: {
				type: Boolean,
				default: false
			},
			version: {
				type: String,
				default: undefined
			},
			hideLogo: {
				type: Boolean,
				default: false
			},
			hideSocialMediaLinks: {
				type: Boolean,
				default: false
			},
			socialMediaLinks: {
				type: Array as PropType<SocialMediaLink[]>,
				default: () => defaultSocialMediaLinks
			}
		}
	});

	const MixinsDeclaration = mixins(Props, customizable(config));

	@Component({
		inheritAttrs: false,
		components: {
			FooterBar,
			SocialMediaLinks
		}
	})
	export default class FooterBar extends MixinsDeclaration {
		locales = locales;

		arrowTopIcon = mdiArrowUp;

		get a11yComplianceLabel(): string {
			const complianceLabel = locales[this.a11yCompliance];

			return locales.a11yLabel(complianceLabel);
		}

		get extendedMode(): boolean {
			return Boolean(this.$slots.default);
		}

		get logoSize(): LogoSizeEnum {
			return this.$vuetify.breakpoint.smAndDown ? LogoSizeEnum.SMALL : LogoSizeEnum.NORMAL;
		}

		get footerLinksMapping(): LinkItem[] {
			if (this.linkItems) {
				return this.linkItems;
			}

			const linksMapping = [
				{
					text: locales.sitemapLabel,
					to: this.sitemapRoute,
					hidden: this.hideSitemapLink
				},
				{
					text: locales.cguLabel,
					to: this.cguRoute,
					hidden: this.hideCguLink
				},
				{
					text: locales.cookiesLabel,
					to: this.cookiesRoute,
					hidden: this.hideCookiesLink
				},
				{
					text: locales.legalNoticeLabel,
					to: this.legalNoticeRoute,
					hidden: this.hideLegalNoticeLink
				},
				{
					text: this.a11yComplianceLabel,
					to: this.a11yStatementRoute,
					hidden: this.hideA11yLink
				}
			];

			return linksMapping.filter((item) => !item.hidden);
		}

		getLinkComponent(item: LinkItem): string {
			return item.href ? 'a' : 'RouterLink';
		}

		scrollToTop(): void {
			window.scrollTo({
				top: 0,
				behavior: 'smooth'
			});
		}
	}
</script>

<style lang="scss" scoped>
	@import '@cnamts/design-tokens/dist/tokens';

	$white: #fff;

	// Use deep selector to style user content as well
	.vd-footer-bar.theme--dark :deep() {
		background-color: $vd-grey-darken-20 !important;

		.vd-footer-bar-links a {
			color: $white !important;
		}

		p,
		.primary--text {
			color: rgba($white, .6) !important;
		}

		svg {
			fill: $white !important;
		}
	}

	.vd-footer-bar-links :deep() {
		a {
			transition: .15s;
			text-decoration: none;
			padding-top: 1px; // Add top padding to account for bottom border
			border-bottom: 1px solid transparent;

			&:hover,
			&:focus {
				border-color: currentColor;
			}
		}

		p {
			padding: 1px 0;
		}
	}
</style>
