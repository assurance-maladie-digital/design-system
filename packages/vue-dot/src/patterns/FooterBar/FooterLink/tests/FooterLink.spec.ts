import Vue from 'vue';

import { Wrapper } from '@vue/test-utils';

import { mountComponent } from '@/tests';
import { html } from '@/tests/utils/html';

import FooterLink from '../';
import FooterBar from '../../index';

let wrapper: Wrapper<Vue>;

describe('FooterLink', () => {
	it('renders correctly', () => {
		wrapper = mountComponent(FooterLink, {
			stubs: ['RouterLink']
		});

		expect(html(wrapper)).toMatchSnapshot();
	});
});
