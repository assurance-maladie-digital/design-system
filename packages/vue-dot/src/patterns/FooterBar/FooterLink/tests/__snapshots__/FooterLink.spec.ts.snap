// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterLink renders correctly 1`] = `
<div class="d-flex">
  <routerlink to="[object Object]"></routerlink>
</div>
`;
